components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/AbstractOTPAuthenticatorConstants.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.util;
+
+/**
+ * Constants related to OTP Authenticator.
+ */
+public class AbstractOTPAuthenticatorConstants {
+
+    public static final int DEFAULT_OTP_LENGTH = 6;
+    public static final int DEFAULT_OTP_RESEND_ATTEMPTS = 5;
+    public static final long DEFAULT_OTP_VALIDITY_IN_MILLIS = 300000;
+    public static final String USERNAME = "username";
+    public static final String AUTHENTICATORS = "authenticators=";
+    public static final String IDF_HANDLER_NAME = "IdentifierExecutor";
+    public static final String LOCAL_AUTHENTICATOR = "LOCAL";
+    public static final String IS_LOGIN_ATTEMPT_BY_INVALID_USER = "isLoginAttemptByInvalidUser";
+    public static final String INVALID_USERNAME = "invalidUsername";
+    public static final String FAILED_LOGIN_ATTEMPTS_CLAIM_URI = "http://wso2.org/claims/identity/failedLoginAttempts";
+
+    // OTP generation.
+    public static final String OTP_NUMERIC_CHAR_SET = "9245378016";
+    public static final String OTP_ALPHA_NUMERIC_CHAR_SET = "KIGXHOYSPRWCEFMVUQLZDNABJT9245378016";
+    public static final String RESEND = "resendCode";
+    public static final String CODE = "OTPCode";
+    public static final String OTP_TOKEN = "otpToken";
+    public static final String OTP = "otp";
+    public static final String OTP_RESEND_ATTEMPTS = "otpResendAttempts";
+
+    // Query params.
+    public static final String AUTHENTICATORS_QUERY_PARAM = "&authenticators=";
+    public static final String RETRY_QUERY_PARAMS = "&authFailure=true&authFailureMsg=authentication.fail.message";
+    public static final String ERROR_USER_ACCOUNT_LOCKED_QUERY_PARAMS =
+            "&authFailure=true&authFailureMsg=user.account.locked";
+    public static final String ERROR_USER_CLAIM_NOT_FOUND_QUERY_PARAMS =
+            "&authFailure=true&authFailureMsg=user.claim.not.found";
+    public static final String ERROR_USER_RESEND_COUNT_EXCEEDED_QUERY_PARAMS =
+            "&authFailure=true&authFailureMsg=resent.count.exceeded";
+    public static final String SCREEN_VALUE_QUERY_PARAM = "&screenValue=";
+    public static final String UNLOCK_QUERY_PARAM = "&unlockTime=";
+    public static final String MULTI_OPTION_URI_PARAM = "&multiOptionURI=";
+    public static final String RECAPTCHA_PARAM = "&reCaptcha=true";
```

Comment:
The name of the constant does not give a hit that it contains `&` and `=`. Would be good to just keep the param key and append the `&` and `=` when you are building it.